documents/coredocument.go

```diff
@@ -416,16 +433,14 @@ func (cd *CoreDocument) signingRootTree(docType string) (tree *proofs.DocumentTr
 
 	// create the signing tree with data root and coredoc root as siblings
 	tree = NewDefaultTreeWithPrefix(ConvertToProofSalts(cd.Document.CoredocumentSalts), SigningTreePrefix, compactProperties(SigningTreePrefix))
-	prefixProp := NewLeafProperty(SigningTreePrefix, compactProperties(SigningTreePrefix))
-
 	err = tree.AddLeaves([]proofs.LeafNode{
 		{
-			Property: prefixProp.FieldProp(DataRootField, binary.LittleEndian.Uint32(compactProperties(DataRootField))),
+			Property: NewLeafProperty(fmt.Sprintf("%s.%s", SigningTreePrefix, DataRootField), append(compactProperties(SigningTreePrefix), compactProperties(DataRootField)...)),
```

Comment:
avoid little endian encoding, and letting the precise proof library take care of that.